src/app/group/timeline/timeline.component.ts

```diff
@@ -9,18 +9,45 @@ import { TASKS } from './mock-tasks'
 })
 
 export class TimelineComponent implements OnInit {
-  tasks = TASKS;
-  selectedTask: Task;
+  tasks = TASKS
+  selectedTask: Task
+  currentDate: Date = new Date()
+  fiveDaysInMillisec: number = 5 * 24 * 60 * 60 * 1000
 
   onSelect(currentTask: Task): void {
-    if(this.selectedTask === currentTask){
-      this.selectedTask = null;
-    } else {
-      this.selectedTask = currentTask;
+    this.selectedTask = currentTask
+  }
+
+  isOldTask(task: Task): boolean {
+    let taskDate: Date = new Date(task.date)
+
+    if (((this.currentDate.getTime() - taskDate.getTime()) > this.fiveDaysInMillisec)
+       && task != this.selectedTask) { //so that we don't try to paint tasks in blue and yellow simultaneously
+      return true
     }
   }
 
-  constructor() { }
+  constructor() {
+
+    let todayTime = new Date(this.currentDate.toLocaleDateString()).getTime()
```

Comment:
It would be better to use const if you don't plan to change todayTime?

Reply:
changed to const